### PR TITLE
BUG: Fixed accessor for Categorical[Datetime]

### PR DESCRIPTION
**Pull request by TomAugspurger**
Closes https://github.com/pandas-dev/pandas/issues/19468

No release note, since this was only a regression on master.

cc @jschendel.

### CODE REVIEW

**Comment by jschendel** [comment]
Thanks, LGTM

File: pandas/tests/series/test_datetime_values.py [hunk 1]
```diff
@@ -259,6 +259,14 @@ def f():
 
             pytest.raises(com.SettingWithCopyError, f)
 
+    def test_dt_namespace_accessor_categorical(self):
+        # GH 19468
+        dti = pd.DatetimeIndex(['20171111', '20181212']).repeat(2)
+        s = pd.Series(pd.Categorical(dti), name='foo')
```
Comment (jreback) [inline pandas/tests/series/test_datetime_values.py]: can you not use ``pd.`` here where we don't elsewhere

**Review by jreback — COMMENTED** [review]
lgtm other than minor point.

**Comment by jreback** [comment]
lgtm. merge on green.

**Pull request merged**